samples/apps/copilot-chat-app/webapi/Controllers/ChatHistoryController.cs

```diff
@@ -0,0 +1,189 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using SemanticKernel.Service.Model;
+using SemanticKernel.Service.Skills;
+using SemanticKernel.Service.Storage;
+
+namespace SemanticKernel.Service.Controllers;
+
+/// <summary>
+/// Controller for chat history.
+/// </summary>
+[ApiController]
+[Authorize]
+public class ChatHistoryController : ControllerBase
+{
+    private readonly ILogger<ChatHistoryController> _logger;
+    private readonly ChatSessionRepository _chatSessionRepository;
+    private readonly ChatMessageRepository _chatMessageRepository;
+    private readonly PromptSettings _promptSettings;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="ChatHistoryController"/> class.
+    /// </summary>
+    /// <param name="logger">The logger.</param>
+    /// <param name="chatSessionRepository">The chat session repository.</param>
+    /// <param name="chatMessageRepository">The chat message repository.</param>
+    /// <param name="promptSettings">The prompt settings.</param>
+    public ChatHistoryController(
+        ILogger<ChatHistoryController> logger,
+        ChatSessionRepository chatSessionRepository,
+        ChatMessageRepository chatMessageRepository,
+        PromptSettings promptSettings)
+    {
+        this._logger = logger;
+        this._chatSessionRepository = chatSessionRepository;
+        this._chatMessageRepository = chatMessageRepository;
+        this._promptSettings = promptSettings;
+    }
+
+    /// <summary>
+    /// Create a new chat session and populate the session with the initial bot message.
+    /// </summary>
+    /// <param name="chatParameters">Object that contains the parameters to create a new chat.</param>
+    /// <returns>The HTTP action result.</returns>
+    [HttpPost]
+    [Route("chat/create")]
+    [ProducesResponseType(StatusCodes.Status201Created)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<IActionResult> CreateChatAsync(
```

Comment:
Let's try to be super consistent with our terms. What's the term we want to use here - "Chat" or "ChatSession"? It feels like "chat session" is clearer, so consider "CreateChatSessionAsync". Same comment for all methods in this class.